pkg/config/crd/crdwatcher.go

```diff
@@ -0,0 +1,273 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package crd provides the implementation of store interfaces to
+// use kubernetes custom resource definitions (CRDs).
+//
+// Note that this is a transient approach. After all of the tasks are done,
+// this package will be replaced by more directly connected to the Istio's
+// config model.
+package crd
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/ghodss/yaml"
+	"github.com/golang/glog"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/dynamic"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp" // for k8s config for GKE
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"istio.io/mixer/pkg/config/store"
+)
+
+type resource struct {
+	version string
+	spec    interface{}
+}
+
+// Mapping from the resource name to the resource spec.
+type resourceMap map[string]*resource
+
+type watcher struct {
+	namespace string
+	stoppers  []func()
+
+	mu       sync.Mutex
+	data     map[string]string
+	listener store.Listener
```

Comment:
This should be `[]store.Listener`, right? As implemented if I call `RegisterListener` twice I only get notified on the last listener.